include/boost/gil/image_processing/filter.hpp

```diff
@@ -60,6 +69,71 @@ void blur(
     box_filter(src_view, dst_view, kernel_size, anchor, true, option);
 }
 
+
+namespace detail
+{
+template <typename SrcView, typename DstView>
+void impl_median(SrcView const& src_view, DstView const& dst_view, std::size_t kernel_size)
```

Comment:
Please, rename `impl_median` to `median_impl` or even `filter_median_impl`.

The `_impl` should always be  a postfix, not prefix.